strawberry_django_plus/utils/query.py

```diff
@@ -1,21 +1,34 @@
+from __future__ import annotations
+
 import functools
-from typing import List, Optional, Set, Type, TypeVar
+from typing import TYPE_CHECKING, TypeVar, cast
 
 from django.contrib.auth import get_user_model
-from django.contrib.contenttypes.models import ContentType
 from django.db.models import Exists, F, Model, Q, QuerySet
 from django.db.models.functions import Cast
 from strawberry_django.utils import is_async
 
-from .typing import TypeOrIterable, UserType
+if TYPE_CHECKING:
+    from django.contrib.auth.models import AbstractUser
+
+    from .typing import TypeOrIterable
+
+
+try:
+    from django.contrib.contenttypes.models import ContentType
+
+    has_contenttypes = True
+except (ImportError, RuntimeError):
+    has_contenttypes = False
```

Comment:
I think this is better because `has_contentypes = True` doesn't imply in `ContentType` existing for the type checking. With this you can add a `if ContentType is not None` and pyright/mypy will understand that it exists inside the block

```suggestion
try:
    from django.contrib.contenttypes.models import ContentType
except (ImportError, RuntimeError):
    ContentType = None
```

Reply:
pyright doesn't like this. It will cause a hell of errors and I could only fix it by has_contenttypes

Reply:
If you have a better idea, please tell me. I found no other solution

Reply:
the problem is:
Union[ContentType, None]

It is non-intuitive that ContentType = None causes here trouble, but this is the case